### PR TITLE
Adding some "Projects usually have these" things

### PR DESCRIPTION
**Pull request by jacobangel**
Lemme know what sucks and what's ok!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,26 +45,27 @@ import { Route, Switch } from 'react-router';
...
 
 ### Installation
 
 `npm install --save react-router-transition`
 
 ### Usage
 
-`RouteTransition` requires a few props:
-- `pathname`: the key signifying the transitionable route, most typically the pathname (required)
+#### `<RouteTransition />`
```
Comment (maisano) [inline README.md]: i actually want to tuck away `RouteTransition`, and only expose it as a low-level wrapper for `TransitionMotion`, should `AnimatedSwitch` or `AnimatedRoute` not be enough.

File: README.md [hunk 2]
```diff
@@ -45,26 +45,27 @@ import { Route, Switch } from 'react-router';
...
-- `atLeave`: an object of interpolatable style values for a route that is unmounting (required)
-- `atActive`: interpolatable style values for a route that has mounted (required)
 - `mapStyles`: an optional function to transform styles that aren't 1:1 (e.g. animating `translateX` or other values of `transform`)
 - `runOnMount`: a boolean to signal whether or not to run the transition on initial `RouteTransition` mount
-
-and supports a couple optional props:
-- `className`: applies to the wrapper component
-- `style`: applies to the wrapper component
+- [`className`]: Optional. `className` is defined on the wrapper component
+- [`style`]: Optional. Styles are applied to the wrapper component
```
Comment (maisano) [inline README.md]: i think i removed the `style` prop

File: README.md [hunk 3]
```diff
@@ -95,6 +96,47 @@ const styles = this.props.location.action === 'POP'
...
+<AnimatedRoute
+  path="/show-me"
+  atEnter={{ opacity: 0 }}
+  atLeave={{ opacity: 0 }}
+  atActive={{ opacity: 1 }}
+  component={MyComponent}
+/>
+```
+
+#### `<AnimatedSwitch />`
```
Comment (maisano) [inline README.md]: let's order these `AnimatedSwitch`, `AnimatedRoute`, `RouteTransition`.

File: README.md [hunk 4]
```diff
@@ -95,6 +96,47 @@ const styles = this.props.location.action === 'POP'
...
+      atLeave={{ opacity: 0 }}
+      atActive={{ opacity: 1 }}
+      key={location.key}
+      location={location}
+    >
+        <Route exact path="/" component={Home} />
+        <Route exact path="/about/" component={About}/>
+    </AnimatedSwitch>
+  )} />
+</div>
```
Comment (maisano) [inline README.md]: this example should just be:

```jsx
<AnimatedSwitch
  atEnter={{ opacity: 0 }}
  atLeave={{ opacity: 0 }}
  atActive={{ opacity: 1 }}
>
  <Route exact path="/" component={Home} />
  <Route path="/about" component={About} />
</AnimatedSwitch>
```

Comment (maisano) [inline README.md]: the wrapper will inline the wrapping `Route` component and handle location-passing.

File: README.md [hunk 5]
```diff
@@ -45,26 +45,27 @@ import { Route, Switch } from 'react-router';
...
 
 ### Usage
 
-`RouteTransition` requires a few props:
-- `pathname`: the key signifying the transitionable route, most typically the pathname (required)
+#### `<RouteTransition />`
+The primary component for declaring a route transition.
+
+`<RouteTransition />` requires a few props:
+- `pathname` (required): the key signifying the transitionable route, most typically the `location.pathname`.
```
Comment (maisano) [inline README.md]: `pathname` has been removed in favor of using `key` in the child element.

File: __tests__/RouteTransition-test.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import RouteTransition from '../src/RouteTransition';
+
+test('RouteTransition mounts', () => {
+  // Render a checkbox with label in the document
```
Comment (maisano) [inline __tests__/RouteTransition-test.js]: snip the comment?

File: src/AnimatedSwitch.js [hunk 7]
```diff
@@ -63,9 +63,8 @@ class AnimatedSwitch extends React.Component {
 // inject location as a prop so we can listen for changes
 const RouteWrapper = props => (
   <Route
-    children={({ location }) => (
-      <AnimatedSwitch location={location} {...props} />
-    )}
+    children={({ location }) =>
+      <AnimatedSwitch location={location} {...props} />}
```
Comment (maisano) [inline src/AnimatedSwitch.js]: i think i prefer this with the parens–implicit return expressions that span multiple lines are weird looking without them.

**Comment by jacobangel** [comment]
We'd want to revisit this from scratch.

**Pull request closed without merging**